### ISSUE TITLE
has doesn't work with sparse arrays

### ISSUE THREAD

**Issue by mxdubois** [mariocasciaro/object-path#73]
Hey, I've noticed that `has` throws an error when I try to check a nested path in a sparse array. I believe it should return `false` in this case.

Example:
`objectPath.has({ foo: [undefined, undefined, undefined] }, 'foo.0.bar')` 

Throws:
```
 Uncaught TypeError: Cannot convert undefined or null to object(…)
 objectPath.has @ index.js:136
```

**Comment by vfernandestoptal** [comment]
Got the same error on an object property that has a null/undefined value, like
```
objectPath.has({ a: null }, 'a.b')
```
```
TypeError: Cannot convert undefined or null to object
    at hasOwnProperty (native)
    at Function.objectPath.has (.../object-path/index.js:136:81)
```

So I think this is more general than just array.

**Comment by mariocasciaro** [comment]
Fixed in 0.11.2

**Comment by vfernandestoptal** [comment]
thanks! :+1:
